extension/commercial_data/README.md

```diff
@@ -0,0 +1,416 @@
+# Commercial Data Extension
+
+The Commercial Data API extension provides an interface for discovering, ordering and using commercial data in the openEO API. 
+
+- Version: **0.1.0**
+- Stability: **experimental**
+
+**Note:** This document only documents the additions to the specification.
+Extensions can not change or break existing behavior of the openEO API.
+
+## Overview of the workflow
+
+All the available datasets provided by a backend are listed on the `/collections` endpoint. The collections are normally freely accessible. This extensions adds capabilities for providing collections that are not free of charge and require purchasing data products that can thereupon be used in processing. Commercial data collections usually allow purchasing small subsets of the data (products), for example a single observation of an area.
+
+Therefore, the client must have an ability to search the available products that match their desired criteria and inspect their metadata to decide whether the products should be purchased.
+
+The client can then create an order for the desired products. Because of the financial cost of purchasing the data, a separate endpoint for confirming the execution of that specific order should be implemented.
+
+When the order is completed, the user should be able to access the data using the `collection-id` at the temporal and spatial location of the purchased products.
+
+### Collection discovery
+
+A backend should add general information about a commercial data collection to the `/collections` and `/collections/{collection-id}` endpoints, the same as with freely available collections. Only the metadata about the entire dataset needs to be provided, not about the specific data products that a user has already purchased. 
```

Comment:
I'm not sure this is needed, because the temporal and spatial availability of products is available in the information about orders.

However, this might be an argument for implementing `collections/{collection_id}/orders` as mentioned in https://github.com/Open-EO/openeo-api/pull/447#discussion_r924801576.